core/src/test/java/org/elasticsearch/node/internal/InternalSettingsPreparerTests.java

```diff
@@ -42,6 +44,8 @@
 import static org.hamcrest.Matchers.*;
 
 public class InternalSettingsPreparerTests extends ESTestCase {
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
```

Comment:
> The rule sucks because it has side effects, if we write another test that uses it, it can easily have some bogus leftover state from a previous test method?

I don't think so. I think that @Rule is just like @Before in that it's processed once before each test method thus setting the state to empty before each test method runs. I'll research.